text/0000-variadic-tuples.md

```diff
@@ -0,0 +1,1006 @@
+- Feature Name: variadic_tuples
+- Start Date: 2019-08-16
+- RFC PR: [rust-lang/rfcs#0000](https://github.com/rust-lang/rfcs/pull/0000)
+- Rust Issue: [rust-lang/rust#0000](https://github.com/rust-lang/rust/issues/0000)
+
+# Summary
+
+[summary]: #summary
+
+Tuples types are ordered set of type, but users can only use tuple with fixed arity.
+
+This RFC aims to allow the use of a _variadic tuple_ which is a tuple with an arbitrary arity.
+
+# Motivation
+
+[motivation]: #motivation
+
+## Arbitrary tuple arity support
+
+Currently, when a user wants to either use or add behavior to tuples, he writes an impl for each tuple arity.
+For easier maintenance, it is usually done with a `macro_rules` and implemented up to 12 arity tuple. (see [ `Hash` implementation in `std`](https://github.com/rust-lang/rust/blob/master/src/libcore/hash/mod.rs)).
+
+Variadic tuple will provide several benefits considering trait implementation for tuple or using variadic tuples:
+
+- Implementations will be easier to write
+- Implementations will be easier to read and maintain
+- The compiler will compile implementation only for required tuple arity
+
+# Guide-level explanation
```

Comment:
This… really isn't a very good "guide-level" explanation. It explains everything, but not in the way that "guide-level" explanations merit.

It douses the user with far more detail and jargon than necessary, and it doesn't explain the feature as if it already exists; it explains it as it *doesn't*, which isn't really what we need.

Will add a bunch of comments on things to improve, but really, I feel like this section needs heavy revision.